src/blob/core.rs

```diff
@@ -285,9 +285,6 @@ where
             let record = Record::deleted(key)?;
             self.write(record).await?;
             let res = self.index.mark_all_as_deleted(key)?;
-            if on_disk {
```

Comment:
We should be able to dump immediately if we are low on memory. I suggest to pass additional boolean flag `dump_immediately`, that controls this behavior and on the Bob side set it according to `index_memory_limit` to avoid exceeding it

Reply:
Isn't it better to leave it to bob? We can add method like `trim` or `free_possible_resources` to pearl public API and make bob use it if limits are reached. Adding flag for only one operation seems like ad hoc solution

Reply:
On a call, we decided to add a flag to the function